src/app/Core/api.service.ts

```diff
@@ -0,0 +1,20 @@
+export interface ApiResponse<T>{
+    apiStatus: number;
+    apiProp?: T;
+}
+
+export async function ApiService(apiUrl: string): Promise<ApiResponse<Record<string, unknown>[]>> {

```

Comment:
Sorry, i should have explained it in more detail. You can use the generic type T for functions too:
`function ApiService<T extends Object>(url: string): Promise<ApiResponse<T>>`
So no conversions necessary on calling the service: `const foo: FooDTO = ApiService('fooUrl)`
I think `Object` is the more appropriate type for a generic json response.

Reply:
Yes, I understood that. But when I do it like this I get the error "can not find name T" in the return type Promise<ApiResponse<T>>.

Reply:
? This works:
```
interface ApiResponse<T extends Object> {
    status: number;
    prop?: T
}
async function getFromApi<T extends Object>():Promise<ApiResponse<T>> {
    const response = await fetch('someUrl');
    const x: ApiResponse<T> = {
        status: response.status,
        prop: await response.json()
    }
    return x;
}

interface MyData {
    foo: number;
}

async function test() {
    const response: ApiResponse<MyData> = await getFromApi();
    if(response.prop !== undefined) {
        const foo = response.prop.foo; // typed as number
    }
}
```

Reply:
Hint: you can also use a shorter form: `const response = await getFromApi<MyData>();` in the test function.

Reply:
Ah I forgot the Type behind the function. But I have to use the type Record<string, unknown>[]  instead of Object, because its a more precise description. (linter)

Reply:
Okay. I can not use Record<string, unknown>, because its not matching the DTO. But when I use Object, the linter is complaining. Should I deactivate this rule?

Reply:
Perhaps Record<string, any>? or T extends any would also be possible (that's what comes from the json() parser.

Reply:
Hint: You can disable a linter rule for a specific line

Reply:
What does the linter say to Object? You can also use T extends {} | [{}]

Reply:
Can I remove the "extends X" completely? than it would work. 
Using any is also forbitten by the linter. 
Disable a linter rule for one line seems so hacky ?! 😄

Reply:
linter to Object: Don't use `Object` as a type. The `Object` type actually means "any non-nullish value", so it is marginally better than `unknown`.

Reply:
Disabling a linter rule for a line is not hacky if you know what you do. Your json() gives you any as a result so you know that and you can disable the linter rule to adapt to it.

Reply:
So the linter is only for hinting you that something in your code might be wrong, which is in most cases true.

Reply:
I would not remove the extend, only T means unknown, but you know that your api has a specific format: Object, so you should type it accordingly.

Reply:
And the linter complains about {} in the same way.

Reply:
So, i would prefer {} and disable the rule for the relevant lines. If your api returns an array not encapsulated in an object (i don't know) it would be {} | [{}]

Reply:
Done.